egs2/TEMPLATE/asr1/pyscripts/feats/dump_km_label.py

```diff
@@ -170,7 +177,19 @@ def dump_label(
                 )
 
                 for idx, utt in enumerate(utt_ids):
-                    lab = apply_kmeans(feats[idx][: feats_lens[idx]].numpy())
+                    if "feature_maxlen" in kwargs and feats_lens[idx] > kwargs.get(
```

Comment:
The following only deals with assigning labels to feature segments. But the feature was already extracted from `reader.get_feats` above. It may not solve the long-recording feature extraction issue.

Reply:
Yup - sorry looks like I pushed the wrong file(s). Will update